src/Internal/Plutus/Types/CurrencySymbol.purs

```diff
@@ -98,6 +99,10 @@ mkCurrencySymbol byteArr
   | otherwise =
       scriptHashFromBytes byteArr $> CurrencySymbol byteArr
 
+-- | Bypasses the usual checks of mkCurrencySymbol: use with extreme care!
+mkCurrencySymbolUnsafe :: ByteArray -> CurrencySymbol
+mkCurrencySymbolUnsafe = CurrencySymbol
+
```

Comment:
This function is not needed anymore.

Reply:
Is there a reason not to use the constructor?

Reply:
No reason really. I thought that the hidden constructor had some purpose, and having a function with `unsafe` in the name makes it clear that you're doing something dangerous.